Source/Renderer/Buffer.js

```diff
@@ -1,20 +1,54 @@
 /*global define*/
 define([
         '../Core/defaultValue',
+        '../Core/defined',
         '../Core/defineProperties',
         '../Core/destroyObject',
-        '../Core/DeveloperError'
+        '../Core/DeveloperError',
+        '../Core/IndexDatatype',
+        './BufferUsage'
     ], function(
         defaultValue,
+        defined,
         defineProperties,
         destroyObject,
-        DeveloperError) {
+        DeveloperError,
+        IndexDatatype,
+        BufferUsage) {
     "use strict";
 
     /**
      * @private
      */
-    var Buffer = function(gl, bufferTarget, sizeInBytes, usage, buffer) {
+    var Buffer = function(context, bufferTarget, typedArrayOrSizeInBytes, usage) {
+        var gl = context._gl;
+        var sizeInBytes;
+
+        if (typeof typedArrayOrSizeInBytes === 'number') {
```

Comment:
Instead of waiting, I suggest we clean up this hackery (my doing many years ago) now by using an `options` object where one of two properties must be defined `sizeInBytes` or `arrayBuffer`.  The call to `bufferData` doesn't care, but it will clean up the Cesium API where we rarely allow a single variable to accept more than one type.